org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/utils/ClientLogHandler.java

```diff
@@ -0,0 +1,58 @@
+package org.eclipse.lsp4xml.utils;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.logging.Handler;
+import java.util.logging.LogRecord;
+import java.util.logging.SimpleFormatter;
+
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+import org.eclipse.lsp4j.services.LanguageClient;
+
+/**
+ * LogHandler
+ */
+public class ClientLogHandler extends Handler{
+  private LanguageClient languageClient;
+
+  public ClientLogHandler(LanguageClient languageClient) {
+    this.languageClient = languageClient;
+  }
+
+  @Override
+  public void publish(LogRecord record) {
+    if(languageClient == null) {
+      return;
+    }
+    StringBuilder sb = new StringBuilder();
```

Comment:
you missed yesterday's comments about moving all that formatting (including the time) into its own testable method